accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+
+E.g. of the output:
+
+```
+NAME         OS-IMAGE                              KERNEL-VERSION        CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+master0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+worker0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+```
+
+The output is presented in the "CaaS Platform > cluster name" section.
+
+### Removing nodes from the cluster
+
+A node can also be removed from the cluster:
+
+- temporarily (cordoned off): using `kubectl` from the Management Node via Salt.
+- permanently: by invoking `skuba node remove <nodename>` via Salt
+
+Example state:
+
+```
+remove_node:
+      cmd.run:
+        - name: skuba node remove <nodename>
+```
+
+NOTE: Other `skuba` options can be implemented in the future (e.g. resetting and reconfiguring a node).
+
+### Cluster upgrade
+
+All the cluster updates must be handled by `skuba`.
+
+SUSE Manager can check and notify the user if an update for the cluster is present by triggering `skuba cluster upgrade plan` on the Management Node (using Salt) and check if a new output is available in a cron fashion (e.g. Taskomatic job).
+
+SUSE Manager will then interface with the Management node and upgrade every master in the cluster first:
+
+`skuba node upgrade apply --target <master-node-ip> --user <user> --sudo`
+
+and then every worker:
+
+`skuba node upgrade apply --target <master-node-ip> --user <user> --sudo`
+
+This can be done by supplying the command to the Management Node via Salt.
+
+### CVE Auditing
+
+By leveraging the existing feature of CVE Auditing, SUSE Manager can detect and notify the user whether the cluster is running a vulnerable image. When SUSE Manager rebuilds the image and patches it, SUSE Manager can also guide the user in modifying the Kubernetes deployment to pull the updated image (in case `imagePullPolicy:` is not `Always`) or do a `kubectl rolling-update`.
+
+## Future work
+
+### Deploy OpenStack Cloud
+
+The approach described in this RFC can be further reused for deploying and managing an OpenStack Cloud using AirShip, given that `skuba` is similar to `airshipctl`:
+
+- provision nodes to be part of the cloud
+- tag nodes and bootstrap them to be part of the cloud
+- manage the lifecycle of the cloud
+
+### Run other products in the cluster
+
+It would also be possible, after that the cluster has been deployed, to deploy other SUSE products on top of the cluster (e.g. SUSE OpenStack Cloud, SUSE Enterprise storage): every product would have a deployment descriptor on top of CaaS Platform, and SUSE Manager would only have to deploy CaaS Platform and then the deployment of the product.
+
+In this case, SUSE Manager would be the central point of deployment for other products based on CaaS Platform and show the health of every cluster installed, while offering the possibility of managing and upgrading the cluster.
```

Comment:
I like to think this case as follows:

- CaaSP is initialized
- CaaSP deploys a SUSE Application Lifecycle operator on top of the cluster
- This SUSE Application Lifecycle defines certain set of CRD's that will allow anyone to deploy SOC, SES or any other combination of products on top of the cluster, managing their lifecycle as well (product upgrades...). As long as you have access and rights to access the cluster with a given `kubeconfig` file, you can create/change these custom resources.

In this case, SUMA could talk to the Kubernetes API, reading these CRD's (or altering them), so it could be the common view showing all products deployed on every cluster, and maybe even being able to alter those deployments, like upgrading SOC or SES by changing a given custom resource that the SUSE Application Lifecycle manager will act upon.

Reply:
Agreed indeed, added and rephrased the paragraph.